src/index.js

```diff
@@ -138,19 +145,19 @@ function getHandlers(set, props) {
     set(state => {
       // if the same DOM el as previous just return state
       if (state.el === el) return state
-      // if new DOM el clean up old DOM
-      if (state.el && state.el !== el) cleanUp(state.el)
+
+      // store event attached DOM el for comparison, clean up, and re-attachment
+      const newState = { ...state, el }
+
+      // if new DOM el clean up old DOM and reset cleanUpTouch
+      if (state.el && state.el !== el && state.cleanUpTouch) {
+        newState.cleanUpTouch = state.cleanUpTouch()
```

Comment:
instead of setting up `newState`, do a `transientState`?
```
const transientState = {}
if (state.el && state.el !== el && state.cleanUpTouch) {
  state.cleanUpTouch()
  transientState.cleanUpTouch = null
}
...
return {...state, ...transientState}
```
Then we can be more conistent and clear with using `state` for checking and `transientState` for updating?